torch/csrc/jit/python/init.cpp

```diff
@@ -1940,6 +1940,15 @@ void initJITBindings(PyObject* module) {
             ss << self;
             return ss.str();
           })
+      .def(py::pickle(
+          [](const FunctionSchema& self) { // __getstate__
+            std::stringstream ss;
+            ss << self;
+            return py::str(ss.str());
+          },
+          [](py::str schema) { // __setstate__, note: no `self` argument
+            return parseSchema(schema);
+          }))
```

Comment:
Why did this PR cause us to need pickle support for FunctionSchema?

Reply:
I added FakeScriptObject to the metadata of nodes that refer to ScriptObjects so that we can use it to get the schema of the call_torchbind operations. In some places, we deepcopy the metadata which failed because FunctionSchema is not pickle-able.